### DISCUSSION TITLE
Home Assistant entities are not discovered in any way while unit is correctly working.

### DISCUSSION THREAD

**Discussion by Ombruz** [revk/xESP32-Faikout#523] [Q&A] [answered]
Good Evening everyone.

I have a Daikin unit with Faikin board correctly installed and working. WiFi signal is stable, the board web interface is reacheable pointing to the right IP address and it's possible to send commands (on/off/temp etc.).

In MQTT Explorer the corresponding topics are generated and it's possible to make operations on the AC unit manually publishing the right topics to the broker. 
HA is correctly working and updated.
In the Faikin setting page I switched to ON the options "Home Assistant" and "HA extra switches" together and alternatively, but I can't detect anything on Home Assistant. No buttons, sensors or anything, no entities or devices.
Mosquitto autodiscovery is on, other sensors and switches (tuya, tasmota etc.) are correctly autodetected. Turning it off does of course have no effect.
Is there any way to force HA to specifically search the Faikin or its entities? I dig into many forum and sources, nothing so far helped.
Alternatively could someone post the topics corresponding to the Faikin entities, or any Yaml code to manually build the mqtt buttons and sensor entities?

Many thanks for any kind of help!

**Comment by revk** [comment]
Faikin posts `homeassistant/` config messages one `ha` is turned on, and HA just picks those up!

Anyone else got any ideas?

**Comment by wurmtal868** [comment]
I was under the impression that I had to install an integration like Tasmota, to deal with the MQTT messages from Faikin.
It took me while that the standard MQTT integration list Faikin as an MQTT-HVAC device manufactured by RevK.
Maybe that's the case here as well.

**Reply by revk** [reply]
Yes, we do standard HA integration. We *also* support the old Daikin integration, but that is not as good.

**Reply by Ombruz** [reply, marked as answer]
Dear revk, first of all thanks for the incredible work you made with the faikin, which now works like a charm and fully integrated.
The solution was just changing in HA the discovery topic from homeassistant to homeassistant/
Hope it will help others too!

**Reply by revk** [reply]
What in faikin settings? If so I need to change default.

**Comment by frischfish** [comment]
I have to pick up this topic again.
My device suddenly disappeared in HA and is not detected automatically.
It works fine via the webinterface.
I changed the topic according to Ombruz, but that didn't help.
I can see Faikout in MQTT sending data all the time, so mqtt connection is working.
Any ideas?

**Reply by revk** [reply]
Check the messages set at startup, normally `home assistant/`...

**Reply by frischfish** [reply]
Stupid question...where?

**Reply by revk** [reply]
In whatever tools you have to see MQTT messages.

**Reply by frischfish** [reply]
I cannot find anything refering to my device name nor faikout, assuming that you are talking about startup / reboot of faikout

**Reply by revk** [reply]
Yeh, the Faikout should send messages starting `homeassistant/` when it starts and is connected to the a/c.

**Reply by frischfish** [reply]
OK looks like it doesn't. Is it possible to debug by it is not sending something?

**Reply by revk** [reply]
Check the `ha` settings is on in the Faikout settings.

Also, this is definitely connected to aircon?

**Reply by frischfish** [reply]
I was waiting for this question.
It is definitly connected to the aircon, I can control it via the web interface.
haenable is on.

**Reply by revk** [reply]
OK, then it definitely should send that message to configure home assistant via MQTT.

**Reply by frischfish** [reply]
So we are back at my initial problem :)

**Reply by revk** [reply]
So I may need to add some sort of debug to the web page perhaps.

I'll haw a look at that.

**Reply by revk** [reply]
Beta issued, shows HA sent or not at top of settings/Faikout page

**Reply by frischfish** [reply]
![002188](https://github.com/user-attachments/assets/fb241175-de2d-490c-8274-41969d8d4644)

**Reply by revk** [reply]
OK, that is a clue! I'll have to add some more debug!

**Reply by revk** [reply]
Ok try next beta in a few minutes

**Reply by frischfish** [reply]
"Note, HA config not yet sent (send pending) (a/c not talking)"

-> I have switched the a/c on via faikout:

"Note, HA config not yet sent (send pending)"


Rebooted, with a/c on:

"Note, HA config not yet sent (send pending)"

**Reply by revk** [reply]
OK, more clues... leave with me a bit.

**Reply by revk** [reply]
There really is not a lot that can break this, it has to do 10 poll loops and then sends. A communication error is only way to really break that.

Do you have any MQTT messages saying `noack`, `badhead`, `mismatch`, `badsum`, etc?

**Reply by frischfish** [reply]
I can see /error/DGKlimaanlage/comms/

{"protocol":"S21","timeout":true}
{"protocol":"S21","cmd":"FX","mismatch":true,"data":"024700000000004703"}
{"protocol":"S21","timeout":true}
{"protocol":"S21","cmd":"FU","mismatch":true,"data":"024700000000004703"}

I do get correct /state/.../status/ messages, protocol S21

**Reply by revk** [reply]
Ok sounds like I may need to have `no` settings for `FX` and `FU` - leave with me. Or perhaps recognise these as failures and not send...

**Reply by revk** [reply]
Ok try next beta on this... in a few minutes.

**Reply by frischfish** [reply]
What do you expect?

/comms/ same messages

"Note, HA config not yet sent (send pending)"

**Reply by revk** [reply]
I was hoping it would magically work.

**Reply by revk** [reply]
Ok still timeout logs?

**Reply by revk** [reply]
Adding a new error message to be sure, next beta, in a few minutes

**Reply by frischfish** [reply]
{"protocol":"S21","failed-loop":true}

**Reply by revk** [reply]
Ok good clue, what before that? Sorry this is such a pain to diagnose.

**Reply by frischfish** [reply]
{"protocol":"S21","cmd":"FU","mismatch":true,"data":"024700000000004703"}
{"protocol":"S21","timeout":true}
{"protocol":"S21","failed-loop":true}
{"protocol":"S21","cmd":"FX","mismatch":true,"data":"024700000000004703"}
{"protocol":"S21","timeout":true}
{"protocol":"S21","failed-loop":true}

**Reply by revk** [reply]
Let me look more!

**Reply by revk** [reply]
Ok try next beta in a few minutes.

**Reply by frischfish** [reply]
Note, HA config not yet sent (send pending)
...

{"protocol":"S21","cmd":"FX","mismatch":true,"data":"024700000000004703"}
{"protocol":"S21","timeout":true}
{"protocol":"S21","failed-loop":true,"poll":1}
{"protocol":"S21","cmd":"FU","mismatch":true,"data":"024700000000004703"}
{"protocol":"S21","timeout":true}
{"protocol":"S21","failed-loop":true,"poll":10}
{"protocol":"S21","cmd":"FX","mismatch":true,"data":"024700000000004703"}
{"protocol":"S21","timeout":true}

....

**Reply by revk** [reply]
OK, so I need to work out more.

poll 10 is close but timeout should not exit.

**Reply by revk** [reply]
My code looks flawed, so trying again, next beta in a few minutes

**Reply by frischfish** [reply]
no change, same behavior + messages

**Reply by revk** [reply]
I will review it all and come up with a solution shortly - at least I have a much better idea of the cause.

**Reply by revk** [reply]
OK the original logic worked but the code has moved on a lot and now has issue where a unit does not understand certain message types.

So I am reworking the logic for "on line" which should solve this. Hopefully a bet later today.

**Reply by revk** [reply]
OK please test latest beta.

**Reply by frischfish** [reply]
Thanks a lot:

Note, HA config sent (homeassistant/…)

It is visible in HA!
